pkg/kernel/kernel.go

```diff
@@ -254,7 +253,9 @@ func (k *kernel) runFactories() error {
 				}
 
 				bookLck.Lock()
-				k.addModuleToStage(container.name, module, container.opts)
+				if err = k.addModuleToStage(container.name, module, container.opts); err != nil {
```

Comment:
Do you need to unlock also in case of an error?

Reply:
good catch